platforms/android/library/src/main/java/io/element/android/wysiwyg/EditorStyledTextView.kt

```diff
@@ -138,14 +148,10 @@ open class EditorStyledTextView : AppCompatTextView {
     override fun onAttachedToWindow() {
         super.onAttachedToWindow()
 
-        mentionDetector = if (isInEditMode) null else newMentionDetector()
-
         updateStyle(styleConfig, mentionDisplayHandler)
     }
 
-    override fun onDetachedFromWindow() {
-        super.onDetachedFromWindow()
-
+    fun finalize() {
```

Comment:
The `finalize()` method should be called automatically when the object is about to get garbage collected: https://www.baeldung.com/java-finalize. I can add some docs for this behaviour.

Reply:
Ah I had assumed it was our own function because it was not marked `override fun`. But if it will be called automatically by the garbage collector then I think it's ok.

Reply:
The JNA cleaner does work well enough in this case, and apparently it's preferred over `finalize()` so I'll go with that. I'll have to update the JNA lib, but it should be fine, it's using the same version as we do on EXA.